### DISCUSSION TITLE
Cognito DeleteUser (cognitoidentityprovider) for Go/Golang

### DISCUSSION THREAD

**Discussion by napoleonstanley316** [aws/aws-sdk-go-v2#1860] [Q&A] [answered]
Hello,

I am looking for some help on how to implement/use the DeleteUser function in the cognitoidentityoprovider module.

1.  Are we supposed to download the contents of the api_client.go file, then create a file called client.go and insert the contents of [api_client.go](https://github.com/aws/aws-sdk-go-v2/blob/main/service/cognitoidentityprovider/api_client.go) ?

2. Does anyone have any examples of actually building the request for deleting a user (or any other cognitoidentityprovider functions)?

I am going to attach some screenshots of my code in a word doc.  Hopefully someone will be able to make sense of what I've attempted to do and redirect me where I've gone off course.  I'm new, so it's likely that I'm making this way more complicate than it needs to be. So far, I have not able to follow the documentation in an intuitive manner.

Thanks!

Document of code screenshots:
[DeleteUser Go SDK Function.docx](https://github.com/aws/aws-sdk-go-v2/files/9658315/DeleteUser.Go.SDK.Function.docx)

**Comment by RanVaknin** [comment, marked as answer]
I'm not sure how your project is structured, but you should not make any changes to the source files just to use the SDK.

Here is a code example:

```go
func main() {
	cfg, err := config.LoadDefaultConfig(context.TODO(), config.WithRegion("us-east-1"))
	if err != nil {
		log.Fatalf("unable to load SDK config, %v", err)
	}

	client := cognito.NewFromConfig(cfg)

	deleteUserInput := &cognito.DeleteUserInput{
		AccessToken: aws.String("AccessTokenGoesHere"),
	}
	
	_, err = client.DeleteUser(context.Background(), deleteUserInput)
	if err != nil {
		fmt.Errorf("could not create cognito user: %v", err.Error())
	}
}
```

If you want me to take a look at your code please paste it here in the thread

add three backticks (```) before AND after your code block for github to format the code.
